examples/website/plot/plot-layer/surface-layer.ts

```diff
@@ -56,48 +83,38 @@ export default class SurfaceLayer extends Layer {
     });
     /* eslint-enable max-len */
 
-    gl.getExtension('OES_element_index_uint');
-    this.setState({
-      model: this.getModel(gl)
-    });
+    this.state.model = this.getModel();
   }
 
   updateState({oldProps, props, changeFlags}) {
     if (changeFlags.propsChanged) {
       const {uCount, vCount} = props;
 
       if (oldProps.uCount !== uCount || oldProps.vCount !== vCount) {
-        this.setState({
-          vertexCount: uCount * vCount
-        });
-        this.getAttributeManager().invalidateAll();
+        this.state.vertexCount = uCount * vCount;
+        this.getAttributeManager()!.invalidateAll();
       }
     }
   }
 
-  getModel(gl) {
+  getModel() {
     // 3d surface
-    return new Model(gl, {
+    return new Model(this.context.device as $TODO, {
       id: `${this.props.id}-surface`,
       vs: surfaceVertex,
       fs: fragmentShader,
       modules: [picking],
```

Comment:
FYI, I see three warnings in the JS console:

```
luma.gl: Model(PlotLayer-surface-surface): Ignoring buffer "instancePickingColors" for unknown attribute "instancePickingColors"

luma.gl: Model(PlotLayer-axes-grids): Ignoring buffer "instancePickingColors" for unknown attribute "instancePickingColors"

luma.gl: Model(PlotLayer-axes-labels): Ignoring buffer "instancePickingColors" for unknown attribute "instancePickingColors"
```

It doesn't look like they're directly caused by changes in the PR, and I'm not sure what creates the buffer, so I've left these warnings alone for now.

Reply:
I think we can ignore these for this PR